components/cam/src/physics/crm/samxx/samxx_const.h

```diff
@@ -0,0 +1,224 @@
+
+#pragma once
+
+#include "YAKL.h"
+#include "Array.h"
+#include <iomanip>
+
+using yakl::c::Bounds;
+using yakl::min;
+using yakl::max;
+using yakl::abs;
+using yakl::c::parallel_for;
+using yakl::SArray;
+using yakl::ScalarLiveOut;
+
+typedef double real;
+
+int  constexpr crm_nx     = CRM_NX;
+int  constexpr crm_ny     = CRM_NY;
+int  constexpr crm_nz     = CRM_NZ;
+int  constexpr crm_nx_rad = CRM_NX_RAD;
+int  constexpr crm_ny_rad = CRM_NY_RAD;
+real constexpr crm_dx     = CRM_DX;
+real constexpr crm_dy     = crm_dx;
+real constexpr crm_dt     = CRM_DT;
+
+int  constexpr YES3D = YES3DVAL  ; // Domain dimensionality: 1 - 3D, 0 - 2D
+int  constexpr nx_gl = crm_nx    ; // Number of grid points in X
+int  constexpr ny_gl = crm_ny    ; // Number of grid points in Y
+int  constexpr nz_gl = crm_nz    ; // Number of pressure (scalar) levels
+int  constexpr nsubdomains_x  = 1; // No of subdomains in x
+int  constexpr nsubdomains_y  = 1; // No of subdomains in y
```

Comment:
The nsubdomain variables is another thing we could remove since we're not going to use this for parallelize the CRM.